sqlite_query.c

```diff
@@ -690,3 +732,45 @@ static char * sqlite_text_value_to_pg_db_encoding(sqlite3_stmt * stmt, int stmt_
 		/* There is no UTF16 in PostgreSQL for fast sqlite3_column_text16, hence always convert */
 		return (char *) pg_do_encoding_conversion((unsigned char *) utf8_text_value, strlen(utf8_text_value), PG_UTF8, pg_database_encoding);
 }
+
+/*
+ * Converts int64 from SQLite to PostgreSQL string from 0 and 1 only
+ * s must be allocated with length not less than len + 1 bytes
+ */
+static char *
+int642binstr(sqlite3_int64 num, char *s, size_t len)
+{
+	s[--len] = '\0';
+    do
+		s[--len] = ((num & 1) ? '1' : '0');
+	while ((num >>= 1) != 0);
+    return s + len;
+}
+
+/*
+ * Converts PostgreSQL string from 0 and 1 only to int64 for SQLite
+ */
+sqlite3_int64
+binstr2int64(const char *s)
+{
+    sqlite3_int64 rc = 0;
+	char *bs = (char *)s;
+
+    for (; '\0' != *bs; bs++)
+    {
+		if ('1' == *bs)
+		{
+            rc = (rc * 2) + 1;
+        }
+        else if ('0' == *bs)
+        {
+            rc *= 2;
+        }
+        else
+        {
+            errno = EINVAL;
```

Comment:
I think it is better to raise an error by ereport() instead of returning -1.

Reply:
Really, this will be better. Thanks, @t-kataym ! Fixed in https://github.com/pgspider/sqlite_fdw/pull/83/commits/0c648eacc689fdbf65b7c74a0b1ec8336161bf8b.

Reply:
Thank you for fixing.